src/commands/handlers/registration.js

```diff
@@ -418,6 +432,29 @@ function getSaslAuth(handler) {
     return null;
 }
 
+function handleSaslFail(handler, reason, command) {
+    const event = {
+        reason,
+    };
+
+    if (command) {
+        // Check if we have a server-time
+        const time = command.getServerTime();
```

Comment:
Small nit, we aren't actually checking anything... we are only using server time and else time is set to undefined here so that comment seems to be misleading.

Reply:
I just copied this code and comment from the above handlers